src/site/content/en/blog/codelab-building-a-sidenav-component/index.md

```diff
@@ -0,0 +1,332 @@
+    & > :matches(aside, main) {
+      grid-area: stack;
+    }
+  }
+}
+```
+
+This layout essentially says: Create a named row "stack" with everything in it, 
+and 2 columns in that row, the 2nd of which is named "stack" also. The 1st column 
+should be sized by it's minimal content needs, and the 2nd column can take up the rest. 
+Then, if in a constrained viewport of `540px` or less, put the sidenav and main content elements 
+into the same row and column, resulting in them being on top of each other.
+
+With this as a base, I can now leverage the state of the url bar to toggle the 
+visibility and transition style of the sidenav. Speaking of which, I need to update the 
+HTML to enable CSS to hook into this feature.
+
+```html//0
+<aside>
+<aside id="sidenav-open">
+```
+
+Now the element's ID can match the url hash we'll be setting. Next I added IDs for 
+key elements that control the sidenav.
+
+```html//0
+<a href="#sidenav-open" title="Open Menu" aria-label="Open Menu">
+<a href="#sidenav-open" id="sidenav-button" class="hamburger" title="Open Menu" aria-label="Open Menu">
+```
+
+With the macro layout done, time to style the micro sidenav components.
+
+The `<aside>` has a neat layout too. It has 2 children, a `<nav>` which is the paper like looking 
+component that slides out, and a closing `<a>` link element that sets the url has to '#'. 
+The link is invisible to the right of the paper slide out nav, it's so folks can "click off" the visual 
+component to dismiss it. Here's how I accomplished that with grid.
+
+```css
+#sidenav-open {
+  display: grid;
+  grid-template-columns: [nav] 2fr [escape] 1fr;
+}
+```
+
+I thought the ratio was a really nice touch here, where grid could shine and give a 
+designer a lot of control.
+
+<figure class="w-figure">
+  <video playsinline controls autoplay loop muted class="w-screenshot">
+    <source src="https://storage.googleapis.com/atoms-sandbox.google.com.a.appspot.com/overlay-escape-ratio.mp4">
+  </video>
+</figure>
+
+Next I need to conditionally overlay the main content and persist my position 
+through any document scrolling. This is a great job for `position: sticky` and 
+some `overscroll-behavior`. Go ahead and open the side menu after scrolling the page, 
+or try scrolling the page while the sidenav is open. What do you think?
+
+Here's how I accomplished that:
+
+```css/4-12
+#sidenav-open {
+  display: grid;
+  grid-template-columns: [nav] 2fr [escape] 1fr;
+
+  @media (max-width: 540px) {
+    position: sticky;
+    top: 0;
+    max-height: 100vh;
+    overflow: hidden auto;
+    overscroll-behavior: contain;
+
+    visibility: hidden; /* not keyboard accessible when closed */
+  }
+}
+```
+
+```css/6-8
+#sidenav-open {
+  ...
+
+  @media (max-width: 540px) {
+    ...
+    
+    &:target {
+      visibility: visible;
+    }
+  }
+}
+```
+
+I also only need the open and close buttons when at a mobile layout, let's 
+manage them with CSS and a media query as well.
```

Comment:
I'll just give another example so you have more context about what I'm looking for with my requested changes. In this case the text should be rephased to be a straight-up instruction:

    Add CSS and a media query so that you can manage the open and close buttons when at a mobile layout